include/Matrix.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <unordered_map>
+
+ 
+template <typename T, T defaulVal>
+class Matrix{
+  using Point = std::pair<int, int>;
+  
+  struct PointHash{
+    size_t operator()(const Point& p) const noexcept{
+      size_t hash = 17;
+      const size_t mult = 23;
+      hash = hash*mult + p.first;
+      hash = hash*mult + p.second;
+      return hash;
+    };
+  };
+ 
+  
+  class MatrixElement{
+    Matrix<T, defaulVal>& matrix;
+    
+  public:
+    MatrixElement(Matrix<T, defaulVal>& matrix): 
+          matrix(matrix){};
+    
+    auto& operator=(const T& other){
+      const Point key = std::make_pair(matrix.rowNum,
+            matrix.colNum);
+      if(other == defaulVal)
+        matrix.data.erase(key);
+      else
+        matrix.data[key] = other;
+      return *this;
+    }
+    
+    operator T() {
```

Comment:
```suggestion
    operator T() const {
```
Дабы появилась возможность сравнивать `assert(matrix[0][0] == 11);`, ибо дефолтный `operator==` принимает по const ссылке.